src/api/mod.rs

```diff
@@ -2111,11 +2110,10 @@ impl ApiResponse {
         None
     }
 
-    /// Returns the pagination info
-    pub fn pagination(&self) -> Pagination {
+    fn next_pagination_cursor(&self) -> Option<String> {
         self.get_header("link")
-            .and_then(|x| x.parse().ok())
-            .unwrap_or_default()
+            .and_then(pagination::next_cursor)
+            .map(String::from)
```

Comment:
I believe converting to a `String` is because of borrowing? Possibly caused by the sandwitching I mentioned above. So one more reason to figure out if we really need that?

Reply:
Yes, exactly. But the `resp.convert` is not the only problem; also, we are storing the next cursor in a variable that would outlive `resp` even if `resp.convert` would not take ownership of `resp`. I already spent too much time trying to figure out how to avoid a string creating here; maybe later on in the refactoring we can try to get rid of it.